server/fleet/mdm.go

```diff
@@ -1075,3 +1095,72 @@ type MDMCommandResults interface {
 }
 
 type MDMCommandResultsHandler func(ctx context.Context, results MDMCommandResults) error
+
+// DetermineJSONConfigType checks the JSON data to determine if it is an Apple or Android profile.
+// Returns isApple, isAndroid, error.
+func DetermineJSONConfigType(data []byte) (bool, bool, error) {
+	/*
+		Not in this code but endpoint code maybe:
+		Verify that users get an error if they try to upload a profile with a name that already exists (across all platforms). Add Restrictions.json Android profile, then try to add macOS profile (.mobileconfig) that has PayloadDisplayName = Restrictions.
+		Verify that users get an error if they try to upload an Android profile with specific settings that
+		are restricted (specified in Figma).
+	*/
+	type jsonObj map[string]interface{}
+	var profileKeyMap jsonObj
+	err := json.Unmarshal(data, &profileKeyMap)
+	if err != nil {
+		return false, false, fmt.Errorf("Couldn't add. The file should include valid JSON: %s", err.Error())
+	}
+	if len(profileKeyMap) == 0 {
+		return false, false, errors.New("Couldn't add. JSON is empty")
+	}
+	hasTypeKey := false
+	hasPayloadKey := false
+	hasKeysStartingInUpper := false
+	hasKeysStartingInLower := false
+	hasKeysContainingNonLetters := false
+	for k := range profileKeyMap {
+		if k == "" {
+			return false, false, errors.New("empty string is not a valid JSON configuration key")
+		}
+		if k == "Type" {
+			hasTypeKey = true
+		}
+		if k == "Payload" {
+			hasPayloadKey = true
+		}
```

Comment:
Should probably just `continue` if `hasTypeKey` or `hasPayloadKey` is true? Or set `hasKeysStartingInUpper` to true automatically, so you don't loop on the runes for nothing in this case.